compiler/modules/types/core.bal

```diff
@@ -952,7 +952,7 @@ public function split(SemType t) returns SplitSemType  {
 }
 
 public type Value readonly & record {|
-    string|int|float|boolean|() value;
+    string|int|float|decimal|boolean|() value;
```

Comment:
We now use this union in lots of places, both in this file and elsewhere. We should have a definition for it. I think t:SingleValue is best, because it represents a type that is a SingleValue.

```
public type SingleValue ()|boolean|int|decimal|float|string;
```


But then `Value` isn't good. Solve that like this (note `?` at the end):

```
public type OptSingleValue readonly & record {|
   SingleValue value;
|}?;
```

Change ResolvedConst to use SingleValue.

Reply:
@jclark, I did this change locally and it seems a considerable amount of places are changed. So shall I send a separate PR for this part?

Reply:
This change is done seperately in #769